core/iwasm/libraries/lib-pthread/lib_pthread_wrapper.c

```diff
@@ -111,6 +117,7 @@ typedef struct ThreadInfoNode {
         korp_tid thread;
         korp_mutex *mutex;
         korp_cond *cond;
+        korp_sem *sem;
```

Comment:
The new codes added should be controlled by macro WASM_ENABLE_LIB_PTHREAD_SEMAPHORE:
```
#if #ifdef WASM_ENABLE_LIB_PTHREAD_SEMAPHORE != 0
...
#endif
```